entry.go

```diff
@@ -103,8 +103,7 @@ func (entry *Entry) WithError(err error) *Entry {
 
 // Add a context to the Entry.
 func (entry *Entry) WithContext(ctx context.Context) *Entry {
```

Comment:
Why is the receiver a pointer since you dont manipulate it?

Reply:
Keep consistent with Entry.WithError, Entry.WithField and Entry.WithFields methods in the same file.

Reply:
ok

Reply:
That pointer means memory barriers will be different when creating a new Entry. I would argue that the rest of these With* calls should have receiver passed by value.

Reply:
> That pointer means memory barriers will be different when creating a new Entry. I would argue that the rest of these With* calls should have receiver passed by value.

I think having receiver passed by value or pointer are both ok in this case,  just make sure to keep consistent. For the present logic, we create a new Entry inside With* methods and copy some but not all fields from receiver, pointer receiver might be slightly better (less and cleaner code).